test/agent/cmd/networking/tester/network.go

```diff
@@ -0,0 +1,267 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package tester
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"net"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+// TestNetworkingSetupForRegularPod tests networking set by the CNI Plugin for a list of Pod is as
+// expected
+func TestNetworkingSetupForRegularPod(podNetworkingValidationInput PodNetworkingValidationInput) []error {
+	// Get the list of IP rules
+	ruleList, err := netlink.RuleList(netlink.FAMILY_V4)
+	if err != nil {
+		log.Fatalf("failed to list ip rules %v", err)
+	}
+
+	// Sort the VPC CIDR Range as it will be compared with the
+	// IP Routes for Pods using Secondary ENI IPv4Address
+	sort.Strings(podNetworkingValidationInput.VPCCidrRange)
+
+	// Do validation for each Pod and if validation fails instead of failing
+	// entire test add errors to a list for all the failing Pods
+	var validationErrors []error
+
+	secondaryRouteTableIndex := make(map[int]bool)
+
+	// For each Pod validate the Pod networking
+	for _, pod := range podNetworkingValidationInput.PodList {
```

Comment:
Instead of passing the list of pods from the caller, consider running this as a daemon on every node and let this pull what pods are running on the node this daemon is on, run all validation checks for each pod and report the results. You can scale this framework to large number of nodes/pods this way